src/server/worker/pipeline/common.go

```diff
@@ -19,20 +20,87 @@ import (
 	"github.com/pachyderm/pachyderm/src/server/worker/common"
 	"github.com/pachyderm/pachyderm/src/server/worker/driver"
 	"github.com/pachyderm/pachyderm/src/server/worker/logs"
+
+	"github.com/rjeczalik/notify"
 )
 
-func openAndWait() error {
-	// at the end of file, we open the pipe again, since this blocks until something is written to the pipe
-	openAndWait, err := os.Open("/pfs/out")
-	if err != nil {
-		return err
+// TarFile holder structure
+type TarFile struct {
+	hdr  *tar.Header
+	body []byte
+}
+
+var fileList []TarFile
+var curr TarFile
+var fileCount int
+
+// SkipTarPadding return number of bytes skipped from a tar stream and an EOF
+func SkipTarPadding(r *bytes.Reader) (int64, error) {
```

Comment:
We have our own forked version of tar file. We should check how we handle the padding bytes. It would be good to leverage our implementation.

Reply:
i checked, and it doesn't make any changes to how padding bytes are handled

also, that version is specialized for use in PFS storage internals, while here we need to be using a standard tar, so i don't think it's appropriate to use our fork here